Add HACS compatibility.

prairiesnpr opened the pull request:
This will allow the repo to be added and managed via HACS.  Will address https://github.com/hulkhaugen/hass-bha-icons/issues/2

prairiesnpr committed:
Add HACS compatibility.

hulkhaugen commented:
Cool, great initiative. I'm quite busy these days with my family, so I won't be able to chip in myself. I can merge this request, but I do have a few questions though:

```
resources:
  - type: js
    url: /local/community/hass-bha-icons/hass-bha-icons.js?v=0.1
```
If i'm not mistaken, this only apply to lovelace UI, and not the rest such as in the configuration and customization menus? As opposed to using the `frontend > extra_module_url` in the configuration.yaml. Not a big deal, and I could be wrong, but that's my understanding of how it works. BTW, other plugins have the option to add the resources data to the lovelace.yaml automatically, that's not included here, right?

What about versions? Where are those defined? The v=0.1 doesn't make much sense without having it defined anywhere. I think adding versions to the resources is the old way also, as I believe that HACS keeps it's own version registry, and if no version is defined, it will use the hash of the js-file. Again, this is how i believe it works, I'm not fully sure about it.

The creators of HACS has put a lot of info up here: https://hacs.xyz/docs/publish/start, dunno if you've seen that already.

prairiesnpr commented:
As far as the resources, I agree that is probably confusing and I will remove the version tag, it's not required.  I always put it out of habit, it doesn't do anything other than provide an easy way to prevent cache issues.  You can increment the version number on the link arbitrarily and it will force the browser to pull the updated link and ignore it's cache.

```BTW, other plugins have the option to add the resources data to the lovelace.yaml automatically, that's not included here, right?```  Yes, this will prompt the user to add the required resources to lovelace, there should be no need for them to manually edit anything.

As far as versions, we have a few options, by default, HACS will use the latest commit id as the version.  I think this is sufficient for this project based on the low activity.  If you want to start using the github release feature, HACS will use that for versioning.

prairiesnpr committed:
Added detailed installation instructions and removed version from yaml.

prairiesnpr commented:
Added some images showing the preferred installation process and moved them to a separate file with a link in the main readme.   Let me know if you think that makes the install process clearer.
